lib/spack/spack/relocate_text.py

```diff
@@ -0,0 +1,288 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+"""This module contains pure-Python classes and functions for replacing
+paths inside text files and binaries."""
+
+import re
+from collections import OrderedDict
+from typing import Dict, Union
+
+import spack.error
+
+Prefix = Union[str, bytes]
+
+
+def encode_path(p: Prefix) -> bytes:
+    return p if isinstance(p, bytes) else p.encode("utf-8")
+
+
+def _prefix_to_prefix_as_bytes(prefix_to_prefix) -> Dict[bytes, bytes]:
+    return OrderedDict((encode_path(k), encode_path(v)) for (k, v) in prefix_to_prefix.items())
```

Comment:
General comment (not a request). Since we support Python 3.6+, I don't think we need to use `OrderedDict` anymore

Reply:
True, but I thought the intent was clearer like this